integration/teleterm_test.go

```diff
@@ -0,0 +1,189 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"context"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/gravitational/teleport/api/types"
+	dbhelpers "github.com/gravitational/teleport/integration/db"
+	"github.com/gravitational/teleport/integration/helpers"
+	libclient "github.com/gravitational/teleport/lib/client"
+	"github.com/gravitational/teleport/lib/service"
+	"github.com/gravitational/teleport/lib/srv/db/mysql"
+	api "github.com/gravitational/teleport/lib/teleterm/api/protogen/golang/v1"
+	"github.com/gravitational/teleport/lib/teleterm/api/uri"
+	"github.com/gravitational/teleport/lib/teleterm/clusters"
+	"github.com/gravitational/teleport/lib/teleterm/daemon"
+)
+
+func TestTeletermGatewayCertRenewal(t *testing.T) {
```

Comment:
> This test took 17s on my machine, which is bit heavy. Ideally we could speed it up a bit, but at a minimum we should see if we can make it a parallel test.

I have some ideas on how to make it faster, I have just asked the db access team about feedback on them.

Is it okay if I address that in a separate PR or should wait with merging this? Making the test faster would likely require changes in shared helpers for which I'd want to get some feedback from the db access team.

GCB says the past two runs of this test took ~19s. I think I could cut this time by about 40%

Reply:
I saw 17s as well, but 16.5s of that was just setting up the test clusters. This test doesn't actually need the leaf cluster either, and by commenting out the code that sets up leaf cluster and trust in the setup the time was reduced to ~8 seconds. We should really make that leaf cluster optional.

Reply:
Now that I think about it, we should test Connect gateways with a leaf cluster too. I'll move this test into a separate file in `integration/proxy` and run it as part of `TestALPNSNIProxyDatabaseAccess` so that the setup time is amortized.